cdap-ui/app/cdap/components/Replicator/Detail/ThroughputLatencyGraphs/LatencyGraph/LatencyTooltip.tsx

```diff
@@ -0,0 +1,70 @@
+/*
+ * Copyright © 2020 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+import React from 'react';
+import withStyles, { WithStyles, StyleRules } from '@material-ui/core/styles/withStyles';
+import { IThroughputLatencyData } from 'components/Replicator/Detail/ThroughputLatencyGraphs/parser';
+import { formatLatency } from 'components/Replicator/Detail/ThroughputLatencyGraphs/LatencyGraph/LatencyTable';
+
+const styles = (theme): StyleRules => {
+  return {
+    tooltip: {
+      position: 'absolute',
+      backgroundColor: theme.palette.white[50],
+      boxShadow: theme.shadows[3],
+      padding: '15px',
+      width: '250px',
+    },
+    heading: {
+      marginBottom: '10px',
+    },
+  };
+};
+
+interface ILatencyTooltipProps extends WithStyles<typeof styles> {
+  tooltip: {
+    top: number;
+    left: number;
+    isOpen: boolean;
+    activeData?: Partial<IThroughputLatencyData>;
+  };
+}
+
+const LatencyTooltipView: React.FC<ILatencyTooltipProps> = ({ classes, tooltip }) => {
+  const tooltipData: Partial<IThroughputLatencyData> =
+    tooltip && tooltip.activeData ? tooltip.activeData : {};
+  return (
+    <div
+      className={classes.tooltip}
+      style={{
+        top: tooltip.top,
+        left: tooltip.left,
+        opacity: tooltip.isOpen ? 1 : 0,
+      }}
+    >
+      <div className={classes.heading}>
+        <strong>{tooltipData.formattedTimeRange}</strong>
+      </div>
+      <div>
+        <div>Average latency</div>
```

Comment:
Is there a "standard" dropdown here? Could there be a shared component that takes a couple fields and some list items?

Reply:
the tooltip content designs are all different for each of the graphs unfortunately